### DISCUSSION TITLE
Keine mqtt Daten

### DISCUSSION THREAD

**Discussion by ulrich-lacher** [softwarecrash/Victron2MQTT#17] [Q&A]
Hallo zusammen. Ich habe seit letztem Jahr victron2mqtt und daly2mqtt im Einsatz was auch Monatelang prima funktioniert hat. Seit mehreren Wochen kommt bekomme ich über mehrere Tage keine Daten mehr über mqtt.
Livejson Daten funktionieren sowie auch die Webansicht. Einmal hat es eine Woche gedauert, dann waren die Daten wieder da. Blink Code ist 3x alle 5sec. Auswertung mit iobroker. Mqtt Broker ist mosquitto. Kann da jemand helfen?
Vielen Dank schon mal.

**Comment by all-solutions** [comment]
Hi Ulrich,

3x blinken heißt keine Verbindung zum MQTT-Broker.

Poste doch mal den Inhalt vom livejson hier. Dann kann ich daraus evtl. was ableiten. Ist ja nicht sicherheitskritisch, da da keine personenbezogenen Daten drin stehen.

**Reply by ulrich-lacher** [reply]
{"ESP_Data":{"IP":"192.168.178.67","sw_version":"1.2.1","ESP_VCC":3.315,"Wifi_RSSI":-51,"Flash_Size":4194304,"Sketch_Size":506336,"Free_Sketch_Space":3665920,"Real_Flash_Size":4194304,"Free_Heap":22096,"HEAP_Fragmentation":15,"WS_Clients":0,"Free_BlockSize":18832,"Runtime":6697},"Device_name":"Victron2MQTT","Device_model":"SmartSolar MPPT 150|35","Device_connection":"Connected","Remote_Control_State":true,"Firmware_version_16":"163","Serial_number":"HQ2311DQ9PF","Voltage":53.44,"Battery_current":5.5,"Panel_voltage":127.87,"Panel_power":303,"Operation_state":"Bulk","Tracker_operation_mode":"MPP Tracker active","Off_reason":"None","Current_error":"No error","Load_output_state":"ON","total_kWh":187.36,"today_kWh":0.42,"Max_pow_today":356,"Yesterday_kWh":3.03,"Max_pow_yesterday":997,"Day":94}

**Reply by ulrich-lacher** [reply]
Es betrifft victron2mqtt und daly2mqtt. Beide zeigen das gleiche Verhalten. Mqtt Broker funktioniert auch da ich noch opendtu und tasmota stromzähler über mosquitto zu iobroker schicke.

**Reply by softwarecrash** [reply]
@ulrich-lacher kannst du die mosquito logs bitte noch anhängen?

**Reply by ulrich-lacher** [reply]
[mosquitto.log](https://github.com/softwarecrash/Victron2MQTT/files/15376040/mosquitto.log)

**Reply by ulrich-lacher** [reply]
Firmeware
Daly2MQTT 2.14.0
Victron2MQTT 1.2.1

**Reply by softwarecrash** [reply]
also  laut dem log hast du nicht nur bei daly und victron probleme, opendtu  und tasmota haben alle verbindungsabbrüche in massen, fast immer mit timeouts

**Reply by ulrich-lacher** [reply]
ok, ich habe das Netzwerk zurückgesetzt und mqtt refresh auf 10sec. Im Moment habe ich Daten.
Danke für den Tip.

**Reply by all-solutions** [reply]
Beobachte da auch mal weiter den Mosquitto-Log. Das ist echt ungewöhnlich, dass sich die Geräte alle immer wieder neu verbinden.
Normalerweise verbinden die sich einmal nach z.B. einem Reboot und dann bleibt die Verbindung bestehen. Egal, was da für eine Refresh-Zeit angegeben ist. Gerät und Broker spielen da dann im Hintergrund Ping-Pong, um zu prüfen, ob die Verbindung noch da ist (auch, wenn gerade keine Daten übertragen werden).

**Reply by ulrich-lacher** [reply]
Hallo zusammen. 
Hat jetzt gehalten bis gestern Nachmittag. Seitdem wieder keine Daten. Ich glaube nicht dass es an den Verbindungsproblemen liegt denn von Tasmota und Opendtu bekomme ich weiterhin die Daten geliefert. Habt ihr noch eine Idee?
Grüße Uli

**Reply by softwarecrash** [reply]
verbindungsprobleme sind nunmal verbindungsprobleme, diese solltest du zuerst beheben.
ich kenne den algorhytmus nicht mit dem tasmota und opendtu arbeitet, bei mir verbindet sich der mqtt client neu wenn er die verbindung verliert, was aber passiert wenn er das zb im sekundentakt macht, oder die latenzzeiten ins unermessliche steigen oder andere sachen kann ich nicht sagen. 
falls du eine fritzbox hast, schalte PMF ab.
